### PR TITLE
Sends netdata.public.unique.id (machine GUID) with claim

### PR DESCRIPTION
**Pull request by underhood**
##### Summary
Send Netdata public unique id (machine GUID) when claiming. In case `claiming ID != machine GUID` cloud still knows the GUID.

This will be more important with MVP1 (Streaming support for Cloud)

##### Test Plan
run claiming script with `-verbose` flag. You should see that we send also line:
```
Request to server:
{
    "node": {
        "id": "XXXX",
        "hostname": "timowss"
    },
    "token": "CENSORED",
    "rooms" : [ "YYYY" ],
    "publicKey" : "-----BEGIN PUBLIC KEY-----\nCENSORED\n-----END PUBLIC KEY-----\n",
    "m_guid" : "BBBB"
}
```
`BBBB` can be the same as `XXXX` if -id parameter not given to claiming script.

### CODE REVIEW

**Comment by mfundul** [comment]
Is the cloud aware of the `mGUID` key? Can this cause any troubles if merged quickly?

**Comment by underhood** [comment]
This has been agreed with @hmoragrega. Will not cause trouble. Cloud will ignore new field until update of the cloud. I tested claiming of such agent with production and it was claimed sucessfully.

File: claim/netdata-claim.sh.in [hunk 1]
```diff
@@ -142,6 +142,10 @@ NETDATA_USER=$(get_config_value netdata global "run as user")
 # get the MACHINE_GUID by default
 if [ -r "${MACHINE_GUID_FILE}" ]; then
         ID="$(cat "${MACHINE_GUID_FILE}")"
+        MGUID=$ID
+else
+        echo >&2 "netdata.public.unique.id is not generated yet. Please run agent at least once before attempting to claim. Agent generates this file on first startup."
+        exit 22
```
Comment (mfundul) [inline claim/netdata-claim.sh.in:148]: Take a look at the exit codes at the beginning of the file. It should be documented and also check how the agent code handles the return codes as well. You should check for any collisions or other problems.

Comment (underhood) [reply claim/netdata-claim.sh.in:148]: you are right, thx

**Comment by mfundul** [comment]
Right now the agent will print `"Agent failed to be claimed with an unknown error."` if `netdata.public.unique.id` is not accessible. I would feel more comfortable if you also updated the `claiming_errors[]` array in `claim/claim.c` which also updates `maximum_known_exit_code`. On the other hand, this is going to be a difficult error to reproduce from a running agent.

File: claim/netdata-claim.sh.in [hunk 2]
```diff
@@ -142,6 +144,10 @@ NETDATA_USER=$(get_config_value netdata global "run as user")
 # get the MACHINE_GUID by default
 if [ -r "${MACHINE_GUID_FILE}" ]; then
         ID="$(cat "${MACHINE_GUID_FILE}")"
+        MGUID=$ID
+else
+        echo >&2 "netdata.public.unique.id is not generated yet. Please run agent at least once before attempting to claim. Agent generates this file on first startup."
```
Comment (mfundul) [inline claim/netdata-claim.sh.in:149]: This means that the file is not reable. It's not 100% guaranteed that the file is not generated yet. Maybe change the language from `is not generated yet` to `may not have been generated yet`

**Comment by amoss** [comment]
Why are we sending both? The only purpose for the claiming ID was to allow the unique ID to be replaced if it clashed. If we are changing the system to always send the mGUID and handle clashes on the Cloud then why did it not replace the mGUID?

**Comment by mfundul** [comment]
> 
> 
> Why are we sending both? The only purpose for the claiming ID was to allow the unique ID to be replaced if it clashed. If we are changing the system to always send the mGUID and handle clashes on the Cloud then why did the mGUID not replace the claim ID?

That was my question as well during this process, I preferred the mGUID to be the only thing that identifies the agent. I don't know what the logic is behind having a claiming ID that is separate. What I do know about mGUID though is that if you change it you lose the historical data, since this is a different host now.

**Comment by underhood** [comment]
@amoss this was requested by @hmoragrega as part of Claiming MVP1

**Pull request merged**